cdap-api/src/main/java/io/cdap/cdap/api/app/ApplicationConfigurer.java

```diff
@@ -100,6 +101,13 @@ ScheduleBuilder buildSchedule(String scheduleName, ProgramType programType,
    */
   void schedule(ScheduleCreationSpec scheduleCreationSpec);
 
+  /**
+   * Emit the given {@link Metadata} for the application in user scope
+   *
+   * @param metadata the metadata to emit
+   */
+  void emitAppMetadata(Metadata metadata);
```

Comment:
nit: rename to emitMetadata(). App is redundant since this is in the ApplicationConfigurer